src/shared/ui/Scroll/Scroll.tsx

```diff
@@ -1,19 +1,58 @@
-import React, { FC } from 'react'
+import React, { FC, MouseEvent, useRef } from 'react'
 
 import styles from './Scroll.module.scss'
 
 export type TProps = {
   className?: string
   children: React.ReactNode
+  withManualGrip?: boolean
 }
 
 /**
  * Scrollbar
  * @param {string} className - для дополнительных свойств
  * @param {React.ReactNode} children - элементы внутри компонента
+ * @param {boolean} withManualGrip - флаг, использовать ли ручной захват содержимого для скроллинга
  */
-const Scroll: FC<TProps> = ({ className, children }) => {
-  return <div className={`${styles.scroll} ${className}`}>{children}</div>
+const Scroll: FC<TProps> = ({ withManualGrip = false, className, children }) => {
+  const listRef = useRef<HTMLDivElement>(null)
+  const startX = useRef(0)
+  const startScrollLeft = useRef(0)
+  const isDragging = useRef(false)
+
+  const handleMouseDown = (e: MouseEvent<HTMLDivElement, globalThis.MouseEvent>) => {
+    startX.current = e.clientX
+    if (listRef.current) {
```

Comment:
на рефах проверяем еще сам реф на существование
```suggestion
    if (listRef && listRef.current) {
```